src/components/inputs/LeaseSelectInput.js

```diff
@@ -22,27 +31,86 @@ const LeaseSelectInput = ({
   onBlur,
   onChange,
   placeholder,
-  relatedLeases,
+  leaseHistoryItems,
   value,
 }: Props) => {
-  const getLeaseOptions = (leases: Array<Object>): Array<Object> =>
-    leases
-      .filter((lease) => relatedLeases.find((relatedLease) => lease.id === relatedLease.lease.id) ? false : true)
+  const leaseAttributes = getLeaseAttributes(store.getState());
+  const getHistoryItemOptions = (leases: Array<Object>, plotSearches: Array<Object>,plotApplications: Array<Object>, areaSearches: Array<Object>): Array<Object> => {
+    const filterOutExisting = (item) => leaseHistoryItems.find((leaseHistoryItem) => item.id === leaseHistoryItem.lease.id) ? false : true
+        
+    leases = leases
+      .filter(filterOutExisting)
       .map((lease) => {
+        const stateOptions = getFieldOptions(leaseAttributes, LeaseFieldPaths.STATE)
         return {
           value: lease.id,
-          label: getContentLeaseIdentifier(lease),
+          label: `${getContentLeaseIdentifier(lease) || ''}, ${getLabelOfOption(stateOptions, lease.state) || ''}`,
+          type: 'lease'
         };
       });
 
-  const getLeases = debounce(async(inputValue: string, callback: Function) => {
+    plotSearches = plotSearches
+      .filter(filterOutExisting)
+      .map((plotSearch) => {
+        const maxLength = 24
+        const displayName = plotSearch.name.length > maxLength ? plotSearch.name.substr(0,maxLength) + "..." : plotSearch.name
+        return {
+          value: plotSearch.id,
+          label: `${displayName}, Haku`,
+          type: 'related_plot_application',
+          content_type_model: "plotsearch"
+        };
+      });
+
+    plotApplications = plotApplications
+      .filter(filterOutExisting)
+      .map((plotApplication) => {
+        return {
+          value: plotApplication.id,
+          label: `${plotApplication.application_identifier}, Hakemus`,
+          type: 'related_plot_application',
+          content_type_model: "targetstatus"
+        };
+      });
+
+    areaSearches = areaSearches
+      .filter(filterOutExisting)
+      .map((areaSearch) => {
+        return {
+          value: areaSearch.id,
+          label: `${areaSearch.identifier}, Aluehakemus`,
+          type: 'related_plot_application',
+          content_type_model: "areasearch"
+        };
+      });
+
+      return [...leases, ...plotSearches, ...plotApplications, ...areaSearches].sort((a, b) => (a.label && b.label) && a.label > b.label ? 1 : -1).slice(0,10)
```

Comment:
How does this sorting work, does it sort by labels of the history items? And then what is the slice for?

Reply:
The label is the label of the option of the select element which this function populates with the fetched results.

The slice is for limiting the number of items in the dropdown, but I just realised that the CSS limits the height of the dropdown, and makes it scrollable.